include/envoy/http/codec.h

```diff
@@ -148,11 +148,25 @@ class ConnectionCallbacks {
 };
 
 /**
- * A list of options that can be specified when creating a codec.
+ * http/2 codec settings
  */
-class CodecOptions {
-public:
-  static const uint64_t NoCompression = 0x1;
+struct Http2Settings {
+  uint32_t hpack_table_size_{DEFAULT_HPACK_TABLE_SIZE};
+  uint32_t max_concurrent_streams_{DEFAULT_MAX_CONCURRENT_STREAMS};
+  uint32_t initial_window_size_{DEFAULT_INITIAL_WINDOW_SIZE};
+
+  static const uint32_t MIN_HPACK_TABLE_SIZE = 0;
+  static const uint32_t DEFAULT_HPACK_TABLE_SIZE = 4 * 1024; // from nghttp2
+  static const uint32_t MAX_HPACK_TABLE_SIZE = 16 * 1024 * 1024;
+
+  static const uint32_t MIN_MAX_CONCURRENT_STREAMS = 1;
+  static const uint32_t DEFAULT_MAX_CONCURRENT_STREAMS = 1024;
+  static const uint32_t MAX_MAX_CONCURRENT_STREAMS = 1 << 29; // from MAX_STREAMS
```

Comment:
should we change MAX_STREAMS from source/common/http/http2/conn_pool.h to using this ?